receiver/integrationreceiver/receiver.go

```diff
@@ -0,0 +1,291 @@
+	consumerChain := struct {
+		logs    consumer.Logs
+		metrics consumer.Metrics
+		traces  consumer.Traces
+	}{
+		logs:    r.nextLogsConsumer,
+		metrics: r.nextMetricsConsumer,
+		traces:  r.nextTracesConsumer,
+	}
+
+	receiverConfig, found := config.Receivers[pipeline.Receiver]
+	if !found {
+		return fmt.Errorf("receiver %q not found", pipeline.Receiver)
+	}
+
+	receiverFactory, ok := host.GetFactory(component.KindReceiver, pipeline.Receiver.Type()).(receiver.Factory)
+	if !ok {
+		return fmt.Errorf("could not find receiver factory for %q", pipeline.Receiver.Type())
+	}
+
+	preparedConfig, err := prepareComponentConfig(receiverFactory.CreateDefaultConfig, receiverConfig)
+	if err != nil {
+		return fmt.Errorf("could not compose receiver config for %s: %w", pipeline.Receiver.String(), err)
+	}
+
+	var components []component.Component
+	processors := slices.Clone(pipeline.Processors)
+	slices.Reverse(processors)
+	for i, id := range processors {
+		processorConfig, found := config.Processors[id]
+		if !found {
+			return fmt.Errorf("processor %q not found", id)
+		}
+
+		factory, ok := host.GetFactory(component.KindProcessor, id.Type()).(processor.Factory)
+		if !ok {
+			return fmt.Errorf("could not find processor factory for %q", id.Type())
+		}
+
+		config, err := prepareComponentConfig(factory.CreateDefaultConfig, processorConfig)
+		if err != nil {
+			return fmt.Errorf("could not compose processor config for %s: %w", id.String(), err)
+		}
+
+		params := processor.Settings(r.params)
+		params.ID = component.NewIDWithName(factory.Type(), fmt.Sprintf("%s-%s-%d", r.params.ID, pipelineID, i))
+		params.Logger = params.Logger.With(zap.String("name", params.ID.String()))
+		if consumerChain.logs != nil {
+			logs, err := factory.CreateLogs(ctx, params, config, consumerChain.logs)
+			if err != nil {
+				return fmt.Errorf("failed to create logs processor %s: %w", params.ID, err)
+			}
+			consumerChain.logs = logs
+			components = append(components, logs)
+		}
+		if consumerChain.metrics != nil {
+			metrics, err := factory.CreateMetrics(ctx, params, config, consumerChain.metrics)
+			if err != nil {
+				return fmt.Errorf("failed to create metrics processor %s: %w", params.ID, err)
+			}
+			consumerChain.metrics = metrics
+			components = append(components, metrics)
+		}
+		if consumerChain.traces != nil {
+			traces, err := factory.CreateTraces(ctx, params, config, consumerChain.traces)
+			if err != nil {
+				return fmt.Errorf("failed to create traces processor %s: %w", params.ID, err)
+			}
+			consumerChain.traces = traces
+			components = append(components, traces)
+		}
+	}
+
+	params := r.params
+	params.ID = component.NewIDWithName(receiverFactory.Type(), fmt.Sprintf("%s-receiver", pipelineID))
+	params.Logger = params.Logger.With(zap.String("name", params.ID.String()))
+	receiversCreated := 0
+	if consumerChain.logs != nil {
+		logs, err := receiverFactory.CreateLogs(ctx, params, preparedConfig, consumerChain.logs)
```

Comment:
nit: we could wrap these components constructors into an anonymous function so we could share the switch logic (we will need to duplicate the logic for any new signal type like profiles).


not needed for this PR though

Reply:
I will give a try.